spec-tests/src/test/java/org/candlepin/spec/pools/RefreshPoolsTest.java

```diff
@@ -0,0 +1,877 @@
+            user = adminClient.consumers().createConsumer(user, "admin",
+            owner.getKey(), null, true);
+            ApiClient consumerClient = ApiClients.trustedConsumer(user.getUuid());
+
+            JsonNode entNode = consumerClient.consumers().bindPool(user.getUuid(), pool.getId(), 1);
+            assertEquals(1, entNode.size());
+
+            user = adminClient.consumers().getConsumer(user.getUuid());
+            assertEquals(1, user.getEntitlementCount());
+        }
+
+        // Verify the entitlement count for this pool
+        List<EntitlementDTO> ents = adminClient.pools().getPoolEntitlements(pool.getId());
+        assertEquals(sub.getQuantity(), ents.size());
+
+        // Modify the subscription upstream
+        sub.setQuantity(1L);
+        adminClient.hosted().updateSubscription(sub.getId(), sub);
+
+        refreshPools(adminClient, owner.getKey());
+        pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+        pool = pools.get(0);
+        assertEquals(sub.getQuantity(), pool.getQuantity());
+
+        // Verify the entitlement count has changed
+        ents = adminClient.pools().getPoolEntitlements(pool.getId());
+        assertEquals(1, ents.size());
+    }
+
+    private AsyncJobStatusDTO refreshPools(ApiClient client, String ownerKey) {
+        AsyncJobStatusDTO job = adminClient.owners().refreshPools(ownerKey, true);
+        assertNotNull(job);
+        job = adminClient.jobs().waitForJob(job);
+        assertEquals("FINISHED", job.getState());
+
+        return job;
+    }
+
+    private void compareProducts(ProductDTO expected, ProductDTO actual) {
+        assertEquals(expected.getId(), actual.getId());
+        assertEquals(expected.getCreated(), actual.getCreated());
+        assertEquals(expected.getUpdated(), actual.getUpdated());
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getMultiplier(), actual.getMultiplier());
+        assertEquals(expected.getAttributes(), actual.getAttributes());
+    }
+
+    private void compareBrandings(BrandingDTO expected, BrandingDTO actual) {
+        assertEquals(expected.getProductId(), actual.getProductId());
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getType(), actual.getType());
+    }
+
+    private void compareBrandings(Collection<BrandingDTO> expected, Collection<BrandingDTO> actual) {
+        assertEquals(expected.size(), actual.size());
+        for (BrandingDTO expectedBrand : expected) {
+            boolean found = false;
+            for (BrandingDTO actualBrand : actual) {
+                if (expectedBrand.getProductId().equals(actualBrand.getProductId())) {
+                    compareBrandings(expectedBrand, actualBrand);
+                    found = true;
+                }
+            }
+
+            assertTrue(found, "missing branding " + expectedBrand.getName());
+        }
+    }
```

Comment:
I wasn't sure if there is a better way of comparing the two unordered collections of brandings when we only want to assert against the product id, name, and type. External libraries like AssertJ or apache commons-collections all use the equals method on the BrandingDTO object and that includes the created/updated date time.

Reply:
I like just clearing the created/updated fields since none of the tests care about those fields. If a future test needed to assert on the created and updated fields then they can use the assertThat to compare the elements and use the equals method on the object.

Reply:
That's fine -- just make sure to document that it's happening so it doesn't take anyone by surprise.